examples/transition_callback.rs

```diff
@@ -0,0 +1,50 @@
+//! An example of using state data to propagate events (See issue-17)
+
+#![deny(missing_docs)]
+
+use std::rc::Rc;
+use std::sync::Mutex;
```

Comment:
I don't see why we need all these sync primitives, and I think they may make the example more complex than it should be.

Also, could we insert this into an existing example instead?

Reply:
Yes, moved to dominos example. I also removed the sync primitives. We needed them to have a non-mut reference to self in the callback. Don't see the need to make it mutable at this point in time. If we need it mutable, we can do so in another feature request?

Reply:
I think its probably correct that we don't get mut access in the transition callback, since we shouldn't be mutating the state machine